x/concentrated-liquidity/fees.go

```diff
@@ -199,6 +200,61 @@ func (k Keeper) getInitialFeeGrowthOutsideForTick(ctx sdk.Context, poolId uint64
 	return emptyCoins, nil
 }
 
+// collectFees collects fees from the fee accumulator for the position given by pool id, owner, lower tick and upper tick.
+// Upon successful collection, it bank sends the fees from the pool address to the owner and returns the collected coins.
+// Returns error if:
+// - pool with the given id does not exist
+// - position given by pool id, owner, lower tick and upper tick does not exist
+// - other internal database or math errors.
+// nolint: unused
+func (k Keeper) collectFees(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick int64, upperTick int64) (sdk.Coins, error) {
+	feeAccumulator, err := k.getFeeAccumulator(ctx, poolId)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
+
+	positionKey := formatPositionAccumulatorKey(poolId, owner, lowerTick, upperTick)
+
+	hasPosition, err := feeAccumulator.HasPosition(positionKey)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
+
+	if !hasPosition {
+		return sdk.Coins{}, cltypes.PositionNotFoundError{PoolId: poolId, LowerTick: lowerTick, UpperTick: upperTick}
+	}
+
+	// compute fee growth outside of the range between lower tick and upper tick.
+	feeGrowthOutside, err := k.getFeeGrowthOutside(ctx, poolId, lowerTick, upperTick)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
+
+	// We need to update the position's accumulator to the current fee growth outside
```

Comment:
With the current state of things, we only update the position's fee growth outside accumulator when the position is changed. That is, liquidity is added or removed.

The fee growth outside is updated when users cross ticks by doing swaps. It is possible that other people swap between the time I modify my position and claim rewards.

As a result, when I claim rewards, I want the fee growth outside to be up to date so that I claim all my rewards until now.

Let me know if this makes sense